components/category/Category.js

```diff
@@ -25,52 +23,84 @@ export default function Category({ category, listId }) {
     )
       return;
 
-    updateListItemIndex(destination.index, source.index);
+    updateListItemIndex(destination, source);
   };
 
+  //activeListItems
+  const activeListItems = listItems.filter(
+    (listItem) => listItem.categoryId === category.id && !listItem.disabled
+  );
+  //inactiveListItems
+  const inactiveListItems = listItems.filter(
+    (listItem) => listItem.categoryId === category.id && listItem.disabled
+  );
+
   return (
-    <article>
+    <CategoryWrapper disabled={hasDisabledItems}>
       <CategoryHeadline
         id={category.id}
         extended={isExtended}
         handleClick={() => setIsExtended((oldIsExtended) => !oldIsExtended)}
         listId={listId}
+        disabled={hasDisabledItems}
       >
         {category.name}
       </CategoryHeadline>
       {isExtended && (
         <DragDropContext onDragEnd={handleOnDragEnd}>
-          <Droppable droppableId={category.id}>
+          <Droppable
+            droppableId={
+              hasDisabledItems ? "disabled" + category.id : category.id
+            }
+          >
             {(provided) => (
               <CategoryList
                 ref={provided.innerRef}
                 {...provided.droppableProps}
               >
-                {listItems.map((listItem, index) =>
-                  listItem.categoryId === category.id ? (
-                    <CategoryListItem
-                      key={listItem.id}
-                      id={listItem.id}
-                      index={index}
-                      checked={listItem.checked}
-                      listId={listId}
-                    >
-                      {listItem.name}
-                    </CategoryListItem>
-                  ) : (
-                    ""
-                  )
+                {hasDisabledItems ? (
+                  <>
+                    {inactiveListItems.map((listItem, index) => (
+                      <CategoryListItem
+                        key={listItem.id}
+                        id={listItem.id}
+                        index={index}
+                        checked={listItem.checked}
+                        listId={listId}
+                      >
+                        {listItem.name}
+                      </CategoryListItem>
+                    ))}
+                  </>
+                ) : (
+                  <>
+                    {activeListItems.map((listItem, index) => (
+                      <CategoryListItem
+                        key={listItem.id}
+                        id={listItem.id}
+                        index={index}
+                        checked={listItem.checked}
+                        listId={listId}
+                      >
+                        {listItem.name}
+                      </CategoryListItem>
+                    ))}
+                  </>
                 )}
```

Comment:
This is an unnecessary ternary operator. You can just calculate the relevant list items like this:
```
listItems.filter(
    (listItem) => listItem.categoryId === category.id && listItem.disabled === hasDisabledItems
  )
```